src/SignalR/clients/ts/signalr/src/HandshakeProtocol.ts

```diff
@@ -40,7 +40,20 @@ export class HandshakeProtocol {
             // optional content after is additional messages
             const responseLength = separatorIndex + 1;
             messageData = String.fromCharCode.apply(null, binaryData.slice(0, responseLength));
-            remainingData = (binaryData.byteLength > responseLength) ? binaryData.slice(responseLength).buffer : null;
+
+            if (binaryData.byteLength > responseLength) {
+                remainingData = binaryData.slice(responseLength);
+                // IE 11 does not have .slice() on Uint8Array, we polyfill it with Array.prototype.slice()
+                // which returns an Array object, that doesn't have .buffer, which is an ArrayBuffer
+                // so we need to convert it into a Uint8Array and .buffer that
+                if (remainingData.buffer) {
```

Comment:
If we don't go with the polyfill change, it seems better to create an `isUint8Array` method here and use that.